src/Components/Admin/EnvironmentEditor/environment.js

```diff
@@ -45,30 +47,38 @@ const useStyles = makeStyles((theme) => ({
     },
 }));
 
-export default function EnvironmentEditor() {
+export default function EnvironmentEditor({
+    match: {
+        params: { environmentId },
+    },
+}) {
     const classes = useStyles();
-    const [scenes, setScenes] = useState([
-        {
-            id: "1",
-            name: "Scene 1",
-        },
-        {
-            id: "2",
-            name: "Scene 2",
-        },
-        {
-            id: "3",
-            name: "Scene 3",
-        },
-        {
-            id: "4",
-            name: "Scene 4",
-        },
-        {
-            id: "5",
-            name: "Scene 5",
-        },
-    ]);
+    const [environment, setEnvironment] = useState({});
+    const [scenes, setScenes] = useState([]);
+
+    useEffect(() => {
+        const getEnvironment = async () => {
+            const data = await getScenario(environmentId);
+            setEnvironment(data);
+        };
+
+        if (environmentId) {
+            getEnvironment();
+        }
+    }, [environmentId]);
+
+    useEffect(() => {
+        const getSceneData = async () => {
+            const data = await Promise.all(
+                environment.scene_ids.map(async (id) => getScene(id))
+            );
+            setScenes(data);
+        };
+
+        if (environment.scene_ids) {
+            getSceneData();
+        }
+    }, [environment]);
```

Comment:
I had to define `getEnvironment` and `getSceneData` in the `useEffect` to remove some warnings, but I didn't get the same warning for the `useEffect` in `admin.js`. Anyone have any ideas why this might be the case?

Reply:
Will play around with that warning stuff a little before commenting, but also this can be combined into 1 use effect right? No need for them to be handled separately as far as I can tell

Reply:
I made them separate because I was thinking about the case where we want to start editing scene order. In this case, we would be editing the `environment` state, which should only trigger a GET request for scene data, not a re-trigger for the environment as well. I also tested combining the two together but the scene data never updated. I think it's best if we keep them separate for now